middlewares/authenticate.js

```diff
@@ -26,6 +26,61 @@ const checkRestricted = async (req, res, next) => {
   return next()
 }
 
+/**
+ * Verify if the token in the request is correct and add user data to the request.
+ * @param req
+ * @returns {Promise<Request>}
+ */
+const verifyUserToken = async (req) => {
+  let token = req.cookies[config.get('userToken.cookieName')]
+
+  // Enable Bearer Token authentication for NON-PRODUCTION environments
+  // This is enabled as Swagger UI does not support cookie auth
+  if (process.env.NODE_ENV !== 'production' && !token) {
+    token = req.headers.authorization.split(' ')[1]
+  }
+
+  const { userId } = authService.verifyAuthToken(token)
+
+  // add user data to `req.userData` for further use
+  const userData = await users.fetchUser({ userId })
+  req.userData = userData.user
+
+  return req
+}
+
+/**
+ * Add a refresh token if it satisfies the refresh TTL and add user data to the request.
+ * @param req
+ * @param res
+ * @returns {Promise<{res, isTokenRefreshed: boolean, req}>}
+ */
+const refreshToken = async (req, res) => {
+  const refreshTtl = config.get('userToken.refreshTtl')
+  const token = req.cookies[config.get('userToken.cookieName')]
+  const { userId, iat } = authService.decodeAuthToken(token)
+  const newToken = authService.generateAuthToken({ userId })
+  const rdsUiUrl = new URL(config.get('services.rdsUi.baseUrl'))
+  let isTokenRefreshed = false
```

Comment:
This should be on the top right? This variable being after `newToken` sort of made me confused